src/js/NavigraphNavdataInterface.ts

```diff
@@ -0,0 +1,294 @@
+import { Airport, Airway, Coordinates, NauticalMiles } from "./types"
+import { ControlledAirspace, RestrictiveAirspace } from "./types/airspace"
+import { Communication } from "./types/communication"
+import { DatabaseInfo } from "./types/database_info"
+import { Gate } from "./types/gate"
+import { Approach, Arrival, Departure } from "./types/procedure"
+import { VhfNavaid } from "./types/vhfnavaid"
+import { Waypoint } from "./types/waypoint"
+
+export class NavigraphNavdataInterface {
+  private readonly listener: CommBusListener
+  private queue: CommBusMessage[] = []
+  private eventListeners: Callback[] = []
+  private onReadyCallback: (() => void) | null = null
+
+  private isInitialized = false
+
+  /**
+   * Creates a new NavigraphNavdataInterface
+   *
+   * @remarks
+   * `RegisterCommBusListener` is called during construction. This means that the class must be instantiated once the function is available.
+   */
+  constructor() {
+    this.listener = RegisterCommBusListener(() => {
+      this.onRegister()
+    })
+  }
+
+  /**
+   * Executes a SQL query on the active database
+   *
+   * @remarks
+   * The query must be valid SQL and must be a SELECT query. The query must not specify a special result format.
+   *
+   * @param sql - SQL query to execute
+   * @returns A promise that resolves with the result of the query
+   */
+  public async execute_sql<T>(sql: string, params: string[]): Promise<T[]> {
+    return await this.callWasmFunction("ExecuteSQLQuery", { sql, params })
+  }
+
+  /**
+   * Downloads the navdata from the given URL to the given path
+   *
+   * @param url - A valid signed URL to download the navdata from
+   * @param path - The path to download the navdata to
+   * @returns A promise that resolves when the download is complete
+   */
+  public async download_navdata(url: string, path: string): Promise<void> {
+    return await this.callWasmFunction("DownloadNavdata", { url, path })
+  }
+
+  /**
+   * Sets the download options for all future downloads
+   *
+   * @param batchSize - The number of files to delete or unzip each update (default: 10). This is a performance optimization to avoid blocking the main thread for too long.
+   * @returns A promise that resolves when the function is complete
+   */
+  public async set_download_options(batch_size: number): Promise<void> {
+    return await this.callWasmFunction("SetDownloadOptions", batch_size)
+  }
+
+  /**
+   * Sets the active DFD database to the one at the given path
+   *
+   * @remarks
+   * The path must be a valid path to a folder that contains a DFD navdata database.
+   *
+   * @param path - The path to the folder that contains the DFD navdata
+   * @returns A promise that resolves when the function is complete
+   */
+  public async set_active_database(path: string): Promise<void> {
+    return await this.callWasmFunction("SetActiveDatabase", { path })
+  }
+
+  public async get_database_info(ident: string): Promise<DatabaseInfo> {
```

Comment:
I think you mentioned it already, but types/docs for all of these methods would be really great - but probably a pain to maintain...

Reply:
Im wondering where the best place to put the comments are, because we also have an enum which has every possible function which can be called